### PR TITLE
inserts decisions in the metering VP13

### PR DESCRIPTION
**Pull request by cah-link**

### CODE REVIEW

File: Decisions/scs-0310-v1-datasource_for_metering.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+## Options considered
+
+### using Gnocchi
+
+Gnocchi is already part of the OpenStack cosmos and has its roots there.
+Also, it provides the ability to long term store and offer measures and events.
+
+### using Prometheus
+
+Prometheus is able to store measures only short time.
```
Comment (fkr) [inline Decisions/scs-0310-v1-datasource_for_metering.md:33]: "Prometheus is not suitable as a long term store" fits better imho.

File: Decisions/scs-0310-v1-datasource_for_metering.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Gnocchi is already part of the OpenStack cosmos and has its roots there.
+Also, it provides the ability to long term store and offer measures and events.
+
+### using Prometheus
+
+Prometheus is able to store measures only short time. 
+
+### using the OpenStack services APIs and RabbitMQ
+
+We would have to provide a storage for historical data and whole in on a gnocchi2
```
Comment (fkr) [inline Decisions/scs-0310-v1-datasource_for_metering.md:37]: I fail to comprehend this sentence. Do you want to say that this option misses the storage for long term historical data?

File: Decisions/scs-0310-v1-datasource_for_metering.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# Open questions
+
+* What will be the bottlenecks?
+* What will the granularity of the events meta information?
+
+
+
+# Decision
+
+We will use gnocchi in the first place
```
Comment (fkr) [inline Decisions/scs-0310-v1-datasource_for_metering.md:56]: You state the decision and the options that were considered.
What I'm missing is how we got to this decision based from the options.

File: Decisions/scs-0310-v1-publishing_method_for_metering_data.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+# Introduction
+
+In the past we notice missing events in the telemetry stack of openstack. this results in situations where the Service provider may think that a resource is still in use while the owner shut it down.
+Such missing data makes it difficult for the CSP to provide accurate invoices
+
+# Motivation
+
+we want to fill the gab between measuring resource consumption and having reliable data for invoicing.
```
Comment (fkr) [inline Decisions/scs-0310-v1-publishing_method_for_metering_data.md:16]: Sentences start with a capital letter. 'We[...]

File: Decisions/scs-0310-v1-datasource_for_metering.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+# Design Considerations
+
+In the first place we want to use as much of the already existing infrastructure as possible.
+Also, we want to rely on the already existing APIs as we expect them to be stable over new releases.
+If it shows to be insufficient to use certain APIs because of high costs we consider to use a direct information gathering from the services databases.
+
+## Options considered
+
+### using Gnocchi
```
Comment (fkr) [inline Decisions/scs-0310-v1-datasource_for_metering.md:26]:
```suggestion
### Gnocchi
```

File: Decisions/scs-0310-v1-datasource_for_metering.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+Gnocchi is already part of the OpenStack cosmos and has its roots there.
+Also, it provides the ability to long term store and offer measures and events.
+
+### using Prometheus
```
Comment (fkr) [inline Decisions/scs-0310-v1-datasource_for_metering.md:31]:
```suggestion
### Prometheus
```

File: Decisions/scs-0310-v1-datasource_for_metering.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+Prometheus is able to store measures only short time. 
+
+### using the OpenStack services APIs and RabbitMQ
+
+We would have to provide a storage for historical data and whole in on a gnocchi2
+
+### using the OpenStack services databases
```
Comment (fkr) [inline Decisions/scs-0310-v1-datasource_for_metering.md:39]:
```suggestion
### OpenStack services databases
```

File: Decisions/scs-0310-v1-datasource_for_metering.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+We would have to provide a storage for historical data and whole in on a gnocchi2
+
+### using data from openstack backends like libvirt, qemu, ceph, ovs/ovn and the hosts filesystem
```
Comment (fkr) [inline Decisions/scs-0310-v1-datasource_for_metering.md:43]:
```suggestion
### Gathering data from openstack backends like libvirt, qemu, ceph, ovs/ovn and the hosts filesystem 
```

File: Decisions/scs-0310-v1-publishing_method_for_metering_data.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+# Design Considerations
+We want to be fast and efficient. in implementing the metering project also as in the daily usage of the provided data.
```
Comment (fkr) [inline Decisions/scs-0310-v1-publishing_method_for_metering_data.md:19]: Initially I failed to parse this sentence. 

```suggestion
# Design Considerations

We want to be fast and efficient. This applies to the initial development effort as part of the metering tender VP13 as much as to the result gathering usage data on daily basis.

```

File: Decisions/scs-0310-v1-publishing_method_for_metering_data.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+## Options considered
+
+### building an REST-api
+
+We need to define a REST-Api structure.
+The CSP has to implement requests against this API.
+
+### pushing Daisies
```
Comment (fkr) [inline Decisions/scs-0310-v1-publishing_method_for_metering_data.md:28]: I fail to parse that heading. What does it mean?

**Comment by fkr** [comment]
@cah-link you saw my feedback in the PR?

**Comment by cah-link** [comment]
@fkr Yes. @horazont is working on version 2.0 ;)

**Comment by fkr** [comment]
@horazont this is ready to be reviewed?

File: Decisions/scs-0310-v1-gnocchi-as-metering-database.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+While some small development efforts may be needed,
+to make it fully usable,
+the amount of work is anticipated much less
+than making Prometheus or Influx fit the bill
+(due to backfilling / cardinality scaling constraints),
+let alone rolling a custom implementation.
+
+### Open questions
+
+* What will the granularity of the events meta information?
```
Comment (ppkuschy) [inline Decisions/scs-0310-v1-gnocchi-as-metering-database.md:150]:
```suggestion
* What will be the granularity of the events meta information?
```
Add "be"

File: Decisions/scs-0311-v1-publishing_method_for_metering_data.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+# Open questions
+
+What is necessary to extend the availability of sending to various sinks?
+How does the configuration look like that is needed to push to a sink from the same type that will be already implemented?
+
+
+
+# Decision
+
+We will go the pushing way.
```
Comment (ppkuschy) [inline Decisions/scs-0311-v1-publishing_method_for_metering_data.md:98]: What lead to that decision?
What pros and cons have been discussed?

**Review by ppkuschy — CHANGES_REQUESTED** [review]
Fixed some typos and have open questions regarding the decision process that lead to deciding to use push bashed publishing. (More details would be nice.)

File: Decisions/scs-0310-v1-publishing_method_for_metering_data.md (hunk 4 above)
Comment (artificial-intelligence) [reply Decisions/scs-0310-v1-publishing_method_for_metering_data.md]: also: `gap` not `gab` :wink:

**Comment by fkr** [comment]
@cah-link @horazont from my conversations with @cah-link I was under the impression that within one of these documents the required fields (= metrics) for the json are to be noted. @horazont what is your take on that?

**Review by fkr — APPROVED** [review]
LGTM - **IF** the json for the metrics are to be defined in a different document.

**Comment by fkr** [comment]
The markdown linter does not seem to like sembr. @horazont @maxwolfs

**Comment by maxwolfs** [comment]
> The markdown linter does not seem to like sembr. @horazont @maxwolfs

It's mostly the headings:

> markdownlint-cli2 v0.6.0 (markdownlint v0.27.0)
> Finding: **/*.md !node_modules !.github
> Linting: 37 file(s)
> Summary: 16 error(s)
> Decisions/scs-0310-v1-gnocchi-as-metering-database.md:10 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Introduction"]
> Decisions/scs-0310-v1-gnocchi-as-metering-database.md:27 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Definitions"]
> Decisions/scs-0310-v1-gnocchi-as-metering-database.md:54 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Motivation"]
> Decisions/scs-0310-v1-gnocchi-as-metering-database.md:70 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Design Considerations"]
> Decisions/scs-0310-v1-gnocchi-as-metering-database.md:150:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
> Decisions/scs-0310-v1-gnocchi-as-metering-database.md:157 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Related Documents"]
> Decisions/scs-0310-v1-gnocchi-as-metering-database.md:161 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Conformance Tests"]
> Decisions/scs-03[11](https://github.com/SovereignCloudStack/standards/actions/runs/4963318899/jobs/8882436020?pr=237#step:6:12)-v1-publishing_method_for_metering_data.md:10 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Introduction"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:24 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Definitions"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:44 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Motivation"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:61 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Design Considerations"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:84 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Open questions"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:89 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Decision"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:100 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Related Documents"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:104 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Conformance Tests"]
> Decisions/scs-0311-v1-publishing_method_for_metering_data.md:108 MD0[12](https://github.com/SovereignCloudStack/standards/actions/runs/4963318899/jobs/8882436020?pr=237#step:6:13)/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]

and the DCO:

> Summary
> Commit sha: [f63755f](https://github.com/SovereignCloudStack/standards/pull/237/commits/f63755fd908d06684a9ed156eb2f1f3b7279724b), Author: Jonas Schäfer, Committer: Jonas Schäfer; The sign-off is missing.
> Commit sha: [a5972bb](https://github.com/SovereignCloudStack/standards/pull/237/commits/a5972bb2854919afbdc709b2142e624613352eb8), Author: Jonas Schäfer, Committer: Jonas Schäfer; The sign-off is missing.
> Commit sha: [1771636](https://github.com/SovereignCloudStack/standards/pull/237/commits/1771636b6777f31d167aa4e0fe4df3a2469bb64b), Author: Jonas Schäfer, Committer: Jonas Schäfer; The sign-off is missing.

docs for linting: https://docs.scs.community/community/community/contribute/linting-guide

**Comment by horazont** [comment]
> @cah-link @horazont from my conversations with @cah-link I was under the impression that within one of these documents the required fields (= metrics) for the json are to be noted. @horazont what is your take on that?

My understanding was that this was supposed to be a different document (which I still need to make).

---

I'll look into the linting errors.

**Comment by horazont** [comment]
@fkr Thanks for pointing that out. That should now be fixed I hope.

**Comment by horazont** [comment]
Ok, I'll have to do ugly things now to fix the DCO situation, fasten your seatbelts.

**Comment by horazont** [comment]
@ppkuschy I added words on the rationale, does that address your questions?

**Comment by joshmue** [comment]
Please move the documents into the `Standards` directory. Decision Records go there now, as well.
(The `status-page-comparison.md` file was not created with SCS-0001 in mind yet and should be eventually migrated, too)

File: Standards/scs-0110-v1-gnocchi-as-metering-database.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+- What will be the granularity of the events meta information?
+
+  If we decide to use resource metadata
+  as a place to store slow-changing information
+  (e.g. instance flavors, volume sizes),
+  we need to know what the granularity of that is.
+
+## Related Documents
+
+- SCS-0311-v1
```
Comment (fkr) [inline Standards/scs-0110-v1-gnocchi-as-metering-database.md:159]:
```suggestion
- SCS-0111-v1
```

File: Standards/scs-0111-v1-publishing_method_for_metering_data.md [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+it allows the producer of the data,
+which knows about the interval in which it changes,
+when to provide new data to the consumer.
+In contrast to that,
+a poll-based flow would need the consumer to know about change intervals,
+or alternatively poll in the highest change frequency ever expected.
+
+## Related Documents
+
+- SCS-0310-v1
```
Comment (fkr) [inline Standards/scs-0111-v1-publishing_method_for_metering_data.md:102]:
```suggestion
- SCS-0110-v1
```

**Comment by horazont** [comment]
Hopefully fixed, along with the DCO thing.

**Review by fkr — APPROVED** [review]
whoop whoop whoop

**Pull request merged**